clash-cores/test/Test/Cores/Sgmii/EbTb.hs

```diff
@@ -0,0 +1,117 @@
+module Test.Cores.Sgmii.EbTb where
+
+import Clash.Cores.Sgmii.Common
+import Clash.Cores.Sgmii.EbTb
+import Clash.Hedgehog.Sized.BitVector
+import qualified Hedgehog as H
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Test.Cores.Sgmii.Common
+import Test.Tasty
+import Test.Tasty.Hedgehog
+import Test.Tasty.TH
+import Prelude
+
+-- Check if the output of 'ebTbDecode' is a valid value for a given value from
+-- 'ebTbEncode'.
+prop_ebTbDecodeCheckNothing :: H.Property
+prop_ebTbDecodeCheckNothing = H.property $ do
+  simDuration <- H.forAll (Gen.integral (Range.linear 1 100))
+
+  inp <-
+    H.forAll
+      ( Gen.list
+          (Range.singleton simDuration)
+          (Gen.filterT checkBitSequence genDefinedBitVector)
+      )
+  let simOut = map f inp
+       where
+        f i = isValidDw dw1 && isValidDw dw2
+         where
+          (_, dw1) = ebTbDecode False $ fromCg $ snd $ ebTbEncode False (Dw i)
+          (_, dw2) = ebTbDecode True $ fromCg $ snd $ ebTbEncode True (Dw i)
+
+      expected = replicate simDuration True
+
+  simOut H.=== expected
```

Comment:
This test is too complex for what you are trying to do, the way it is written now it looks like you're testing a sequential component. If you want to make sure you test more, perhaps look at [withTests](https://hackage.haskell.org/package/hedgehog-1.4/docs/Hedgehog.html#v:withTests)